src/OpenTelemetry.Instrumentation.AspNet.Win/Implementation/HttpInListener.cs

```diff
@@ -48,53 +53,75 @@ public override void OnStartActivity(Activity activity, object payload)
 
             if (this.options.RequestFilter != null && !this.options.RequestFilter(context))
             {
+                // TODO: These filters won't prevent the activity from being tracked
+                // as they are fired anyway.
                 InstrumentationEventSource.Log.RequestIsFilteredOut(activity.OperationName);
                 return;
             }
 
+            // TODO: Avoid the reflection hack once .NET ships new Activity with Kind settable.
+            activity.GetType().GetProperty("Kind").SetValue(activity, ActivityKind.Server);
+
             var request = context.Request;
             var requestValues = request.Unvalidated;
 
             // see the spec https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-semantic-conventions.md
             var path = requestValues.Path;
-
-            TelemetrySpan span;
-            if (this.options.TextFormat is TraceContextFormat)
+            activity.DisplayName = path;
```

Comment:
Due to the risk of high-cardinality on the path OTel spec suggests using the HTTP method instead. That said the path is much more useful in general and having an option to add it (more likely the absolute path) will be useful. There are other locations also using the path as DisplayName, but other correctly use the helper that gets the name per OTel spec. BTW, the link on the comment above is broken.

Reply:
Yes incoming requests were still using path as SpanName, hence I used it as activity.DisplayName.
This shouldn't block this PR, as we can (and should) address this separate from this PR.